src/server/string_family.cc

```diff
@@ -550,8 +564,12 @@ void StringFamily::Set(CmdArgList args, ConnectionContext* cntx) {
       sparams.how = SetCmd::SET_IF_NOTEXIST;
     } else if (cur_arg == "XX") {
       sparams.how = SetCmd::SET_IF_EXISTS;
+    } else if (cur_arg == "GET") {
+      std::optional<std::string> prev_val = std::make_optional(std::string(""));
```

Comment:
this is not good. you create a variable and then pass the pointer to it. The stack disappears.